taskcluster/ac_taskgraph/transforms/gradle_optimization.py

```diff
@@ -0,0 +1,32 @@
+from taskgraph.transforms.base import TransformSequence
+
+from ..build_config import get_path, ANDROID_COMPONENTS_DIR
+from ..gradle import get_upstream_deps_for_gradle_projects
+
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def extend_optimization_if_one_already_exists(config, tasks):
+    deps_per_component = get_upstream_deps_for_gradle_projects(ANDROID_COMPONENTS_DIR)
+
+    for task in tasks:
+        optimization = task.get("optimization")
+        if optimization:
+            skip_unless_changed = optimization["skip-unless-changed"]
+
+            component = task["attributes"]["component"]
+            # TODO Remove this special case when ui-test.sh is able to accept "browser-engine-gecko"
+            if component == "browser":
+                component = "browser-engine-gecko"
```

Comment:
I don't understand this workaround. Can you elaborate? What is the browser component and where does it get added? :)

Reply:
I implemented this workaround to keep this existing test working: 

https://github.com/mozilla-mobile/firefox-android/blob/54c1653a88edb76472c9579ed7834ae61b6e98e0/taskcluster/ci/test/kind.yml#L56-L62

Unlike other `ui-test.sh` calls, this time we don't pass an actual Android Component, we just pass `browser`. I might have misunderstood. I based this implementation on the config just above and on this piece of shell script:

https://github.com/mozilla-mobile/firefox-android/blob/54c1653a88edb76472c9579ed7834ae61b6e98e0/android-components/automation/taskcluster/androidTest/ui-test.sh#L85-L97

@AaronMT when we call `ui-test.sh browser`, do we test `browser-engine-gecko`?

Reply:
It was previously `unit-browser-engine-gecko-nightly`, and `ui-browser` `was samples-browser`

Reply:
Okay, I'm keeping the workaround as is then. I'm happy to change it if I misunderstood something.